### DISCUSSION TITLE
Shelly Plus 1PM- SHM zeigt nur maximale Geräteleistung an nicht aktuelle Leistung

### DISCUSSION THREAD

**Discussion by Hoeger53** [camueller/SmartApplianceEnabler#441] [Allgemein]
Hallo, brauche etwas Unterstützung von erfahreneren Usern als ich ;)
Mein neuer SAE läuft und wird im SunnyPortal auch angezeigt. Unter Verbraucherbilanz u Steuerung im SunnyPortal wird mir nur die „Max. Leistungsaufnahme“ angezeigt, die ich im SAE eingestellt habe.
Ich möchte mir mit dem Shelly plus 1pm eigentlich nur den Verbrauch meines Gefrierschrankes anzeigen lassen im Portal. 
Die vielen Hilfeseiten und Infos hier habe ich jetzt rauf und runter gelesen und vieles ausprobiert, leider ohne Erfolg. 
Hat jemand einen Tip für mich?
Vielen Dank

**Comment by camueller** [comment]
Hast Du einen Zähler im SAE konfiguriert?

Wird die von diesem ermittelte Leistung korrekt an der [SEMP-Schnittstelle](https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/SEMP_DE.md) an den SHM übermittelt?

**Reply by Hoeger53** [reply]
Ja, ich habe den Shelly nur als Zähler konfiguriert.  Die Shelly-App zeigt momental 30W Verbrauch an.
Die SEMP-Schnittstelle zeigt folgendes:
<Device2EM>
<DeviceInfo>
<Identification>
<DeviceId>F-04101959-130219600002-00</DeviceId>
<DeviceName>Gefrierschrank SH</DeviceName>
<DeviceType>Freezer</DeviceType>
<DeviceSerial>123456789</DeviceSerial>
<DeviceVendor>Shelly</DeviceVendor>
</Identification>
<Characteristics>
<MaxPowerConsumption>183</MaxPowerConsumption>
</Characteristics>
<Capabilities>
<CurrentPower>
<Method>Measurement</Method>
</CurrentPower>
<Timestamps>
<AbsoluteTimestamps>false</AbsoluteTimestamps>
</Timestamps>
<Interruptions>
<InterruptionsAllowed>false</InterruptionsAllowed>
</Interruptions>
<Requests>
<OptionalEnergy>false</OptionalEnergy>
</Requests>
</Capabilities>
</DeviceInfo>
<DeviceStatus>
<DeviceId>F-04101959-130219600002-00</DeviceId>
<EMSignalsAccepted>false</EMSignalsAccepted>
<Status>On</Status>
<PowerConsumption>
<PowerInfo>
<AveragePower>183</AveragePower>
<Timestamp>0</Timestamp>
<AveragingInterval>60</AveragingInterval>
</PowerInfo>
</PowerConsumption>
</DeviceStatus>
</Device2EM>

**Reply by Hoeger53** [reply]
Ich füge mal die Screenshots an:
![SH-SAE-semp-xml](https://user-images.githubusercontent.com/122628895/212373979-6d3241d6-ae8b-4a0d-b021-f34e78ab1f11.PNG)
![SH-rpc-Shelly-Get Status](https://user-images.githubusercontent.com/122628895/212374015-3849034f-a32d-490c-b4e0-c0062c95a4b1.PNG)
![SH-mit Leistung](https://user-images.githubusercontent.com/122628895/212374063-a95aee1d-db62-4548-bd53-80e28b41ee2c.PNG)

**Comment by camueller** [comment]
Du hast `max Power` auf 183 gesetzt und das ist zufällig auch der Wert der aktuellen Leistungsaufnahme.

**Reply by Hoeger53** [reply]
Ich habe es vorher mit 200, 199 und 188 probiert als max Power und genau diese Werte werden dann immer im Portal aktuell angezeigt. Auch in der SunnyApp unter nichtsteuerbare Verbraucher.

**Reply by Hoeger53** [reply]
Ist denn der SAE Zähler so korrekt konfiguriert, wie in dem Screenshot?

**Reply by camueller** [reply]
> Ist denn der SAE Zähler so korrekt konfiguriert, wie in dem Screenshot?

Schau doch in die Doku - das muss ich auch machen. Und dann schau in die Logs, welcher Wert ausgelesen wird.

**Comment by camueller** [comment]
Der SAE begrenz die gemeldete Leistung auf die gemessene Leistung, d.h. die max. Lesitungsaufname sollte immer grösser sein, als die tatsächliche.

**Reply by Hoeger53** [reply]
Die Doku habe ich natürlich vorher ausführlich gelesen. ;) Vielen Dank für deine superschnelle Reaktion auf mein Problem. 
Ich habe an dieser Steckdose für den Gefrierschrank noch eine FritzDECT angeschlossen. Die zeigt mir im Portal immer einen aktuellen Wert an. Vorhin z.B. über die Fritz 30 W über den SAE wieder den "max Power" (den ich gerade auf 800W gesetzt habe) Wert. Die Shelly App zeigt den gleichen Wert wie die FritzDect.
Ich versuche die Logs noch mal zu überprüfen, bestimmt habe ich etwas übersehen.
Vielen Dank, ich hatte gehofft, dass das Problem vllt schon bekannt ist und es eine schnelle Lösung gibt ;))

**Reply by Hoeger53** [reply]
Problem gelöst, habe beim SAE Zähler für den Shelly plus 1pm bei Parameter: Leistung und beim Pfad für Extraktion: $.apower eingestellt.
Wie für den Shelly 4pm empfohlen. Funktioniert bei mir auch mit dem Shelly Plus 1PM

**Reply by Doenerteller** [reply]
Ich hatte ein ähnliches Problem mit dem Plus 1PM. Bei mir kam gar nix anstatt der maximalen Geräteleistung.
Für den 1PM muss man den Pfad für Extraktion auf "$.switch:0.apower" stellen, dann klappt es. Herausgefunden
durch Rumprobieren.
In der Doku hier steht "$.status:0.apower". Das sollte mal geändert werden. Oder ein Hinweis auf andere mögliche Werte. Vielleicht ist das abhängig von der Firmware des 1PM. Eventuell wurde das von Shelly geändert.

**Comment by camueller** [comment]
Ich habe die Doku diesbezüglich gestern aktualisiert, als ich zusammen mit einem User den fälschlicherweise mit 0 ermittelten Verbauch analysiert habe. Doku ist noch nicht bei Gibthub ...

> In der Doku hier steht "$.status:0.apower". Das sollte mal geändert werden. Oder ein Hinweis auf andere mögliche Werte. Vielleicht ist das abhängig von der Firmware des 1PM.

"mal geändert werden" - ich habe keinen Pull-Request für eine Aktualsierung der Doku von Dir erhalten. Du solltest froh sein, dass es überhaupt Dokumentation zu den Shelly gibt - ich haben keine im Einsatz und kann deshalb nicht sagen, ob und wann sich irgendetwas an deren Firmware ändert. Open-Source-Projekt leben davon, dass alle User einen Beitrag leisten. Wenn sie schon alles umsonst bekommen ...

**Reply by Doenerteller** [reply]
Das sollte kein Vorwurf sein, sondern ein Hinweis. Vielleicht etwas unglücklich formuliert oder verstanden.
Und was ist ein pull request? Eine formale Bitte/Aufforderung etwas zu aktualisieren oder zu tun? Wusste ich nicht, dass man darüber Bitten/Wünsche/Anregungen kommunizieren muss/kann/soll. Daher schrieb ich es einfach in meinen Post.

**Reply by camueller** [reply]
Siehe: https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/Contribute_DE.md

**Comment by Doenerteller** [comment]
Verstehe. Github und seine Eingeweide sind halt für uns Normalos eher Bömische Dörfer...ich probier das mal bei Gelegenheit.
